### PR TITLE
Fix potential double free in link.c

### PR DESCRIPTION
**Pull request by ewertons**
Also added:
- new unit test suite for link.c, which was missing;
- unit test to cover the fix in this PR.

### CODE REVIEW

File: tests/link_ut/link_ut.c [hunk 1]
```diff
@@ -0,0 +1,480 @@
...
+        else
+        {
+            result = 0;
+        }
+    }
+
+    return result;
+}
+
+// 8888888888
```
Comment (vaavva) [inline tests/link_ut/link_ut.c:147]: needed?

File: tests/link_ut/link_ut.c [hunk 2]
```diff
@@ -0,0 +1,480 @@
...
+    *(destination) = *(source);
+
+    return result;
+}
+
+static void umocktypes_free_FLOW_HANDLE(FLOW_HANDLE* value)
+{
+    if (*value != NULL)
+    {
+        // my_gballoc_free(*value);
```
Comment (vaavva) [inline tests/link_ut/link_ut.c:162]: Should this be uncommented?

File: tests/link_ut/link_ut.c (hunk 1 above)
Comment (ewertons) [reply tests/link_ut/link_ut.c:147]: No. Cleaned.

File: tests/link_ut/link_ut.c (hunk 2 above)
Comment (ewertons) [reply tests/link_ut/link_ut.c:162]: Removed.

**Review by vaavva — APPROVED** [review]

**Review by ericwol-msft — APPROVED** [review]

**Pull request merged**